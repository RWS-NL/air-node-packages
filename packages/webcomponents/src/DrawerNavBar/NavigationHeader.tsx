import MUIAppBar from '@material-ui/core/AppBar';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Tabs from '@material-ui/core/Tabs';
import Toolbar from '@material-ui/core/Toolbar';
import clsx from 'clsx';
import { stringify } from 'flatted';
import React, { memo } from 'react';
import { LinkTab } from '../LinkTab';
import { DrawerNavBarProps } from './DrawerProps';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    appBar: {
      zIndex: theme.zIndex.drawer + 1,
      boxShadow: theme.shadows[1],
      transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen
      })
    },
    appBarShift: {
      transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen
      })
    },
    hide: {
      display: 'none'
    },
    toolbar: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      padding: theme.spacing(0, 1),
      // necessary for content to be below app bar
      ...theme.mixins.toolbar
    },
    appBarResize: {
      minHeight: theme.spacing(6),
      maxHeight: theme.spacing(6),
      height: theme.spacing(6),
      [theme.breakpoints.down('sm')]: {
        minHeight: theme.spacing(6),
        maxHeight: theme.spacing(6),
        height: theme.spacing(6)
      }
    },
    tabsBar: {
      justifyContent: 'flex-start',
      paddingLeft: 0
    }
  })
);

export const NavigationHeader = memo(
  ({
    navigationDrawer,
    treeDrawer,
    activeTab,
    onTabChange,
    tabs,
    TabsProps,
    AppBarProps,
    ToolbarProps,
    ...props
  }: DrawerNavBarProps) => {
    const classes = useStyles();

    return (
      <MUIAppBar
        {...props}
        {...AppBarProps}
        position='fixed'
        color='secondary'
        className={clsx(classes.appBar, classes.appBarResize, {
          [classes.appBarShift]: navigationDrawer.open || treeDrawer.open
        })}
      >
        <Toolbar {...ToolbarProps} classes={{ root: clsx(classes.appBarResize, classes.tabsBar) }} variant='dense'>
          <Tabs
            {...TabsProps}
            variant='standard'
            value={activeTab}
            onChange={onTabChange}
            classes={{ indicator: classes.hide }}
          >
            {tabs.map((tab, index) => (
              <LinkTab key={index} label={tab.label} to={tab.to} data-qa={`link-tab-${index}`} />
            ))}
          </Tabs>
        </Toolbar>
      </MUIAppBar>
    );
  },
  (prevProps, nextProps) => {
    if (prevProps.navigationDrawer.open !== nextProps.navigationDrawer.open) return false;
    if (prevProps.treeDrawer.open !== nextProps.treeDrawer.open) return false;
    if (prevProps.activeTab !== nextProps.activeTab) return false;
    if (stringify(prevProps.tabs) !== stringify(nextProps.tabs)) return false;

    return true;
  }
);
