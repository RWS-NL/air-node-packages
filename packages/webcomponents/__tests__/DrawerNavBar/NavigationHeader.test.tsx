import { NavigationHeader } from '@DrawerNavBar/NavigationHeader';
import { drawerNavBarProps, navigationDrawerProps, tabs, treeDrawerProps } from '@Resource/commonDrawerNavBarProps';
import { shallow } from 'enzyme';
import React from 'react';

describe('NavigationHeader', () => {
  describe('Snapshot Testing', () => {
    test('Required Props', () => {
      const navigationHeader = shallow(
        <NavigationHeader
          navigationDrawer={navigationDrawerProps}
          treeDrawer={treeDrawerProps}
          activeTab={0}
          onTabChange={jest.fn()}
          tabs={tabs}
        />
      );
      expect(navigationHeader).toMatchSnapshot();
    });

    test('Optional Props', () => {
      const navigationHeader = shallow(<NavigationHeader {...drawerNavBarProps} />);
      expect(navigationHeader).toMatchSnapshot();
    });
  });
});
