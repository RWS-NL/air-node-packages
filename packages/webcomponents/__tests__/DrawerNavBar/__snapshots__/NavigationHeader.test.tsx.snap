// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationHeader Snapshot Testing Optional Props 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="makeStyles-appBar-1 makeStyles-appBarResize-5 makeStyles-appBarShift-2"
  color="secondary"
  position="fixed"
>
  <WithStyles(ForwardRef(Toolbar))
    classes={
      Object {
        "root": "makeStyles-appBarResize-5 makeStyles-tabsBar-6",
      }
    }
    variant="dense"
  >
    <WithStyles(ForwardRef(Tabs))
      classes={
        Object {
          "indicator": "makeStyles-hide-3",
        }
      }
      onChange={[MockFunction]}
      value={0}
      variant="standard"
    >
      <Memo()
        data-qa="link-tab-0"
        key="0"
        label="Home"
        to="/"
      />
      <Memo()
        data-qa="link-tab-1"
        key="1"
        label="Projectenoverzicht"
        to="/"
      />
      <Memo()
        data-qa="link-tab-2"
        key="2"
        label="Contact"
        to="/"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;

exports[`NavigationHeader Snapshot Testing Required Props 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="makeStyles-appBar-1 makeStyles-appBarResize-5 makeStyles-appBarShift-2"
  color="secondary"
  position="fixed"
>
  <WithStyles(ForwardRef(Toolbar))
    classes={
      Object {
        "root": "makeStyles-appBarResize-5 makeStyles-tabsBar-6",
      }
    }
    variant="dense"
  >
    <WithStyles(ForwardRef(Tabs))
      classes={
        Object {
          "indicator": "makeStyles-hide-3",
        }
      }
      onChange={[MockFunction]}
      value={0}
      variant="standard"
    >
      <Memo()
        data-qa="link-tab-0"
        key="0"
        label="Home"
        to="/"
      />
      <Memo()
        data-qa="link-tab-1"
        key="1"
        label="Projectenoverzicht"
        to="/"
      />
      <Memo()
        data-qa="link-tab-2"
        key="2"
        label="Contact"
        to="/"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
